### PR TITLE
Update travis script & dependencies

### PR DESCRIPTION
**Pull request by YFSHEN97**
The main purpose of this PR is to update the travis script to include the new `test/` folder. The rest are almost all dependency updates. The only changes that matter are `test/e2e/bluegreen_test.go` and `test/service.go`. However, these codes do NOT work as of now; they are still placeholders to make sure the `test/` directory is appropriately set up for `golint` (otherwise `golint` sees an empty `test/` and returns status of non-zero).

### CODE REVIEW

File: test/e2e/bluegreen_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	if err != nil {
+		t.Fatalf("Failed to configure Service with new version: %v: %v", green.Service, err)
+	}
+
+	url := resources.Route.Status.URL.URL()
+	if _, err := pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		t.Logf,
+		url,
+		v1test.RetryingRouteInconsistency(pkgTest.MatchesAllOf(pkgTest.IsStatusOK, pkgTest.MatchesBody("<!DOCTYPE html>"))),
```
Comment (trshafer) [inline test/e2e/bluegreen_test.go]: nit IMHO This doctype doesn't really showcase a successful response

File: test/e2e/bluegreen_test.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	"testing"
+
+	servinge2e "knative.dev/serving/test/e2e"
+
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/serving/test"
+	v1test "knative.dev/serving/test/v1"
+)
+
+func TestBlueGreen(t *testing.T) {
```
Comment (trshafer) [inline test/e2e/bluegreen_test.go]: It looks like this test just validates that the first revision is at 100%. Change the name?

**Review by trshafer — COMMENTED** [review]
Nit: The commit message is "update travis" but IMO the main thing this does is add an e2e test. I would change the commit message to state that a new e2e test folder was created with an e2e test for first revision.

File: test/e2e/bluegreen_test.go (hunk 1 above)
Comment (YFSHEN97) [reply test/e2e/bluegreen_test.go]: The line with doctype would be the first line in the response returned by blue/green, but I guess you're right in that it is a bit too generic. I've changed it to a more characteristic line.

File: test/e2e/bluegreen_test.go (hunk 2 above)
Comment (YFSHEN97) [reply test/e2e/bluegreen_test.go]: It validates that the 2nd (green) version is at 100% in the absence of any policy. I've changed name to `Test100WithouPolicy`.

That being said, this code doesn't work as of now due to a very strange compiling error. I'll still trying to figure out what is causing it.

File: test/e2e/bluegreen_test.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	"testing"
+
+	servinge2e "knative.dev/serving/test/e2e"
+
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/serving/test"
+	v1test "knative.dev/serving/test/v1"
+)
+
+func Test100WithouPolicy(t *testing.T) {
```
Comment (trshafer) [inline test/e2e/bluegreen_test.go]: nit misspelling.

Comment (YFSHEN97) [reply test/e2e/bluegreen_test.go]: Good catch!!

**Review by trshafer — APPROVED** [review]

**Pull request merged**